### PR TITLE
Ixml

### PR DESCRIPTION
**Pull request by ballsteve**
Added an example that shows how to integrate Invisible XML (ixml crate) with Xrust.

The integration would be a lot better, smoother, if the ixml crate had some additional features. In particular, defining a trait for building the parse tree instead of hard-coding the use of indextree. That way, we could supply an intmuttree RNode.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,14 +26,17 @@ harness = false
...
 regex = "1.10.2"
 
 [dev-dependencies]
 criterion = "0.5.1"
 encoding_rs = "0.8.33"
-encoding_rs_io = "0.1.7"
\ No newline at end of file
+encoding_rs_io = "0.1.7"
+#ixml = { package = "earleybird", version = "0.0.1" }
+earleybird = {path = "../earleybird"}
```
Comment (Devasta) [inline Cargo.toml:41]: Could we set this to 

earleybird = {git = "https://github.com/mdubinko/earleybird.git"}

instead? Will automatically resolve the dependency then.

**Comment by Devasta** [comment]
Reviewing at the moment. One test failing for me, xpath_root_desc_or_self_3...

File: Cargo.toml (hunk 1 above)
Comment (ballsteve) [reply Cargo.toml:41]: OK, I'll try that

Comment (ballsteve) [reply Cargo.toml:41]: Done. This has been set in the new-features-2023-11-28 branch.

**Pull request merged**